src/mpid/ch4/shm/ipc/src/ipc_win.c

```diff
@@ -77,19 +78,17 @@ int MPIDI_IPC_mpi_win_create_hook(MPIR_Win * win)
 
     /* Determine submodule availability and IPC type based on buffer type.
      * If no submodule is available, skip initialization; if any submodule
-     * is on, we need always exchange in case remote buffer can be shared by IPC. */
-    MPIDI_IPC_mem_attr_t attr;
-    MPIDI_IPC_get_mem_attr(win->base, ULONG_MAX, shm_comm_ptr->rank, shm_comm_ptr, &attr);
+     * is on, we need always exchange in case remote buffer can be shared
+     * by IPC. We pass MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE as data_sz to
+     * MPIDI_IPC_get_mem_attr becuase for RMA operations we always enable
+     * IPC for buffers larger than 0. */
+    MPIDI_IPC_get_mem_attr(win->base, &attr);
 
```

Comment:
Can we also pass the buffer `extend` down to `MPIDI_IPC_get_mem_attr` and `MPIDI_IPC_xpmem_get_mem_attr`? If the size is zero, XPMEM will set `ipc_type` to NONE.

Reply:
I thought we were not checking the extent in `get_mem_attr` we are doing the check outside in mpi_isend. Should that skip by checking the threshold?